### DISCUSSION TITLE
HSL Compiled Into IPOPT Mex File But HSL Has Been Disabled

### DISCUSSION THREAD

**Discussion by anilvrao** [coin-or/Ipopt#776] [General]
This is a bit of a strange question.

I now have everything working to compile an IPOPT Mex file for Apple Silicon with Mumps and the HSL solvers.  I am, however, getting some really strange behavior with regard to which linear solvers are enabled in the Mex file.  

First, I have HSL disabled in the IPOPT configure command using the flag --without-hsl.  Second, I do not compile the HSL code into my Mex file at all.  Nevertheless, when I am done with the Ipopt compilation and building the Mex file, the HSL linear solvers are still part of the Mex file.  I do not even have an HSL library in /usr/local/lib, so I have no idea why any linking with an HSL library is taking place.  The only linear solver library I have in /usr/local/lib is libcoinmumps.a and libcoinmumps.la.  Finally, when I use the mexIpopt interface built by Enrico Bertolazzi, I have not included linking to an HSL library.  The Mex file builds perfectly fine but, as I said, includes the HSL solvers despite my not including them in the build. 

I am wondering if anyone can help me understand why the HSL solvers are being compiled into my Mex file.  Thanks for the help. 

— Anil

**Comment by svigerske** [comment]
Duplicate of #777.
